doc/java-api.md

```diff
@@ -33,29 +37,30 @@ For instance the JSparkJob uses the JavaSparkContextFactory.
 
 ###Additional Context and Job Types
 
-- `JavaSqlContextFactory` for SQLContext Jobs with `JSqlJob`
-- `JavaStreamingContextFactory` for StreamingContext jobs with `JStreamingJob`
+- `JavaSqlContextFactory` for SQLContext Jobs with `JavaSqlJob`
+- `JavaStreamingContextFactory` for StreamingContext jobs with `JavaStreamingJob`
 
 **NOTE: If you do not use the correct context, the jobs will probably not run correctly, it is imperative you configure
 jobs correctly.**
 
 
-###Configurable Jobs
-Besides the `JSparkJob`, which only configures the return value of `run(...)`, the `JavaSparkJob` exists, which additionally
-allows configuring the return type of `verify(...)`.
+### Convenience Job Interface
+Besides the `JavaSparkJob<D, R>`, which configures the return value of `verify(...)` and input to `run(...)` as well as the
+return value of `run(...)`, SJS offers the `JSparkJob<R>` which defaults to `JavaSparkJob<Config, R>`. You can use this
+interface if you want to pass the spark configuration to `run(...)`.
 
 ```java
-public class JavaConfigurableJob implements JavaSparkJob<String, String> {
+public class JavaHelloWorldJob implements JSparkJob<String> {
 
-    public String run(JavaSparkContext sc, JobEnvironment runtime, String data) {
-        return data;
+    public String run(JavaSparkContext sc, JobEnvironment runtime, Config data) {
+        return "Hi!";
     }
 
-    public String verify(JavaSparkContext sc, JobEnvironment runtime, Config config) {
-        return config.getString("input");
+    public Config verify(JavaSparkContext sc, JobEnvironment runtime, Config config) throws RuntimeException{
+        return ConfigFactory.empty();
```

Comment:
Again, I'm worried that the reader will wonder why the configuration information is not getting passed-through to run in this example. Either, could we add an end-of-line comment saying `// No configuration information needed by run **or** we could have verify check that `! config.getString("name").isEmpty` and then pass that same config object through to run?